projects/fal/src/fal/api.py

```diff
@@ -211,19 +213,37 @@ def wrapper(*remote_args: ArgsT.args, **remote_kwargs: ArgsT.kwargs) -> ReturnT:
             ) from exc.with_traceback(tb)
         finally:
             with suppress(Exception):
-                patch_dill()
+                patch_func()
         return result
 
     return wrapper
 
 
+def _prepare_local_partial_func(
+    func: Callable[ArgsT, ReturnT],
+    *args: ArgsT.args,
+    **kwargs: ArgsT.kwargs,
+) -> Callable[ArgsT, ReturnT]:
+
+    return _prepare_partial_func(func, patch_pickle, *args, **kwargs)
+
+
+def _prepare_remote_partial_func(
+    func: Callable[ArgsT, ReturnT],
+    *args: ArgsT.args,
+    **kwargs: ArgsT.kwargs,
+) -> Callable[ArgsT, ReturnT]:
+
+    return _prepare_partial_func(func, patch_dill, *args, **kwargs)
+
+
 @dataclass
 class LocalHost(Host):
     # The environment which provides the default set of
     # packages for isolate agent to run.
     _AGENT_ENVIRONMENT = isolate.prepare_environment(
         "virtualenv",
-        requirements=[f"dill=={dill.__version__}"],
+        requirements=[f"dill=={dill.__version__}", f"tblib=={tblib.__version__}"],
```

Comment:
isnt `pickling_support.install()` supposed to avoid this?

Reply:
@chamini2 You are absolutely right, esp if we patch the host too. Double checking in a sec...

Reply:
@chamini2 It only provides a way to serialize exceptions, but you still need it on both sides. The difference here and why I didn't need to do the same in #143 is that our remote workers actualy have tblib installed, while local ones don't. E.g. with official fal if you try:

```python
import fal

@fal.function()
def mytest_remote():
    import tblib

    print(f"remote {tblib.__version__}")


@fal.function(host=fal.LocalHost())
def mytest_local():
    import tblib

    print(f"local {tblib.__version__}")


if __name__ == "__main__":
    mytest_remote()
    mytest_local()
```

you'll get:

```bash
(fal-3.11.5) ➜  test_fal python test_151.py
gc:0: ResourceWarning: Object of type google._upb._message.MessageMapContainer is not untracked before destruction
2024-04-01 22:13:36.529 [stdout   ] remote 3.0.0
04/01/2024 22:13:38 [bb0635] [LogSource.BRIDGE] [trace]   Starting the controller bridge.
04/01/2024 22:13:38 [bb0635] [LogSource.BRIDGE] [trace]   Controller server is listening at ('127.0.0.1', 53786). Attempting to start th
e agent process.
04/01/2024 22:13:38 [bb0635] [LogSource.BRIDGE] [trace]   Awaiting agent process of 99884 to establish a connection.
04/01/2024 22:13:38 [bb0635] [LogSource.BRIDGE] [trace]    Starting the isolated process at PID 99884
04/01/2024 22:13:38 [bb0635] [LogSource.BRIDGE] [trace]    Trying to create a connection to ('127.0.0.1', 53786)
04/01/2024 22:13:38 [bb0635] [LogSource.BRIDGE] [trace]   Bridge between controller and the agent has been established.
04/01/2024 22:13:38 [bb0635] [LogSource.BRIDGE] [trace]   Executable has been sent, awaiting execution result.
04/01/2024 22:13:38 [bb0635] [LogSource.BRIDGE] [trace]    Created child connection to ('127.0.0.1', 53786)
04/01/2024 22:13:38 [bb0635] [LogSource.BRIDGE] [trace]    Received the callable at ('127.0.0.1', 53786)
Traceback (most recent call last):
  File "/Users/efiop/git/efiop/ui-comp/test_fal/test_151.py", line 19, in <module>
    mytest_local()
  File "/Users/efiop/.pyenv/versions/fal-3.11.5/lib/python3.11/site-packages/fal/api.py", line 952, in __call__
    return self.host.run(
           ^^^^^^^^^^^^^^
  File "/Users/efiop/.pyenv/versions/fal-3.11.5/lib/python3.11/site-packages/fal/api.py", line 247, in run
    return connection.run(executable)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/efiop/git/efiop/isolate/src/isolate/connections/ipc/_base.py", line 151, in run
    return self.poll_until_result(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/efiop/git/efiop/isolate/src/isolate/connections/ipc/_base.py", line 191, in poll_until_result
    raise prepare_exc(result, stringized_traceback=stringized_traceback)
  File "/Users/efiop/git/efiop/isolate/src/isolate/connections/ipc/agent.py", line 111, in run_client
    result = callable()
^^^^^^^^^^^^^^^
  File "/Users/efiop/git/efiop/ui-comp/test_fal/test_151.py", line 12, in mytest_local
    import tblib
      ^^^^^^^^^^^
ModuleNotFoundError: No module named 'tblib'
```

so remote has `tblib`, while local doesn't. Correct me if I'm wrong, but it looks like it is because we have `isolate` in `agent_requirements` and isolate depends on `tblib`, so happens to install it along the way. But with `LocalHost` we don't install agent requirements (well, that `_AGENT_ENVIRONMENT` is kinda it I guess?) and thus don't have `tblib` installed and so we have to add it here.

Reply:
OK, thanks for the explanation. I think we may need to add it explicitly in the remote requirements if we are now counting on it being installed in them.

Reply:
@chamini2 Makes sense, I'll add it in a separate PR, since that would be for a different project.